packages/mux-uploader/src/progress.ts

```diff
@@ -0,0 +1,167 @@
+    shadowRoot.appendChild(template.content.cloneNode(true));
+
+    this.svgCircle = this.shadowRoot?.querySelector('circle');
+    this.progressBar = this.shadowRoot?.getElementById('progress-bar');
+    this.progressBar?.setAttribute('aria-description', ariaDescription);
+  }
+
+  connectedCallback() {
+    this.setDefaultType();
+    const parent = document.querySelector('mux-uploader');
+
+    parent?.addEventListener('progress', (e) => {
+      // @ts-ignore
+      const percent = e.detail;
+      this.progressBar?.setAttribute('aria-valuenow', `${Math.floor(percent)}`);
+
+      switch (this.getAttribute('type')) {
+        case TYPES.BAR: {
+          if (this.progressBar) this.progressBar.style.width = `${percent}%`;
+          break;
+        }
+        case TYPES.RADIAL: {
+          if (this.svgCircle) {
+            // The closer the upload percentage gets to 100%, the closer offset gets to 0.
+            // The closer offset gets to 0, the more we can see the circumference of our circle. (TD).
+            const offset = this.getCircumference() - (percent / 100) * this.getCircumference();
+
+            this.svgCircle.style.strokeDashoffset = offset.toString();
+          }
+        }
+      }
+    });
+
+    parent?.addEventListener('uploadstart', this.onUploadStart.bind(this));
+  }
+
+  onUploadStart() {
+    this.progressBar?.focus();
+    this.setAttribute('upload-in-progress', '');
+  }
+
+  getRadius() {
+    return Number(this.svgCircle?.getAttribute('r'));
+  }
+
+  getCircumference() {
+    return this.getRadius() * 2 * Math.PI;
+  }
+
+  setDefaultType() {
+    const currentType = this.getAttribute('type');
+
+    if (!currentType) {
+      this.setAttribute('type', TYPES.BAR);
+    }
+
+    if (currentType === TYPES.RADIAL) {
+      if (this.svgCircle) {
+        // strokeDasharray is the size of dashes used to draw the circle with the size of gaps in between.
+        // If the dash number is the same as the gap number, no gap is visible: a full circle.
+        // strokeDashoffset defines where along our circle the dashes (in our case, a dash as long as the
+        // circumference of our circle) begins. The larger the offset, the farther into the circle you're
+        // starting the "dash". In the beginning, offset is the same as the circumference. Meaning, the visible
+        // dash starts at the end so we don't see the full circle. Instead we see a gap the size of the circle.
+        // When the percentage is 100%, offset is 0 meaning the dash starts at the beginning so we can see the circle. (TD).
+        this.svgCircle.style.strokeDasharray = `${this.getCircumference()} ${this.getCircumference()}`;
+        this.svgCircle.style.strokeDashoffset = `${this.getCircumference()}`;
+      }
+    }
+  }
+}
+
+if (!globalThis.customElements.get('mux-uploader-progress')) {
+  globalThis.customElements.define('mux-uploader-progress', Progress);
+}
+
+export default Progress;
```

Comment:
nitpick(blocking): rename to `MuxUploaderProgressElement` for parity with our tacit convention.